### ISSUE TITLE
tls-crypt-v2 broken in OpenVPN 3 Python implementation

### ISSUE THREAD

**Issue by Danrancan** [OpenVPN/openvpn3-linux#71]
**Issue:**
tls-crypt-v2 only partially working. Does not work with autoloader.

**WHAT WORKS:**
Running my openvpn3 using tls-crypt-v2 certs by importing the `configuration.conf`/`auth_configuration.txt` files and then running the session manually with `openvpn3 session-start --config-path /net/openvpn/v3/configuration/05793ae5xde06x54a5xa226x3e77b6e21e45`.

**WHAT DOESN'T WORK**
Running openvpn3 using the systemd autoloader, while implementing the very same `tls-crypt-v2` `configuration.conf`/`configuration.autoload` files.

**MORE INFO**
To be clear, I am trying to start openvpn3 using tls-crypt-v2 with the command `sudo systemctl enable openvpn3-autoload.service`, and it is not connecting at boot, or at all.

**My Logs** look like this:
```
Sep 06 19:37:08 ubuntuserver dbus-daemon[1863]: [system] Successfully activated service 'net.openvpn.v3.netcfg'
Sep 06 19:37:08 ubuntuserver net.openvpn.v3.log[2048]: {tag:7480931689699451224} Network Configuration VERB1: Redirect method: host-route
Sep 06 19:37:08 ubuntuserver net.openvpn.v3.log[2048]: {tag:12882287475413898665} Client -- ERROR --: Failed to parse configuration: ERR_PROFILE_GENERIC: tls_crypt_v2_client_key_parse_error
Sep 06 19:37:08 ubuntuserver net.openvpn.v3.log[2048]: {tag:7480931689699451224} Network Configuration INFO: Cleaning up resources for PID 2200.
Sep 06 19:37:08 ubuntuserver openvpn3-servic[2200]: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
Sep 06 19:37:08 ubuntuserver dbus-daemon[1863]: [system] Rejected send message, 0 matched rules; type="error", sender=":1.21" (uid=116 pid=2200 comm="/usr/lib/aarch64-linux-gnu/openvpn3-linux/openvpn3" label="unconfined") interface="(unset)" member="(unset)" error na>
Sep 06 19:37:09 ubuntuserver net.openvpn.v3.log[2048]: {tag:12882287475413898665} Client -- ERROR --: Failed to parse configuration: ERR_PROFILE_GENERIC: tls_crypt_v2_client_key_parse_error
Sep 06 19:37:09 ubuntuserver net.openvpn.v3.log[2048]: {tag:7480931689699451224} Network Configuration INFO: Cleaning up resources for PID 2200.
Sep 06 19:37:09 ubuntuserver net.openvpn.v3.log[2048]: {tag:12882287475413898665} Client **!! FATAL !!**: Failed to initialize client: Configuration parsing failed: ERR_PROFILE_GENERIC: tls_crypt_v2_client_key_parse_error
Sep 06 19:37:09 ubuntuserver net.openvpn.v3.log[2048]: {tag:15018683109956739556} Session Manager !! CRITICAL !!: Failed communicating with VPN backend: Failed calling D-Bus method Connect: GDBus.Error:net.openvpn.v3.error.client: Configuration parsing failed: ERR_PR>
Sep 06 19:37:42 ubuntuserver net.openvpn.v3.backends[2200]: terminate called without an active exception
```
My **`configuration.conf`** file looks like this:
```
client
proto udp
nobind
remote 111.11.111.11
port 1194
dev tun
dev-type tun
remote-cert-tls server
tls-version-min 1.2
reneg-sec 604800
auth-user-pass
verb 3
push-peer-info
resolv-retry infinite
persist-key
persist-tun
keepalive 10 120

<ca>
-----BEGIN CERTIFICATE-----
...
-----END CERTIFICATE-----
</ca>

<cert>
-----BEGIN CERTIFICATE-----
...
-----END CERTIFICATE-----
</cert>

<key>
-----BEGIN PRIVATE KEY-----
...
-----END PRIVATE KEY-----
</key>

<tls-crypt-v2>
-----BEGIN OpenVPN tls-crypt-v2 client key-----
...
-----END OpenVPN tls-crypt-v2 client key-----
</tls-crypt-v2>
```
My **`configuration.autoload`** file looks like this:
```
{
   "autostart": true,
    "name": "MyConfigurationName",
    "acl": {
        "set-owner": "MyLinuxUserName"
    },
    "tunnel": {
        "ipv6": "no",
        "persist": true,
        "dns-fallback": "google",
        "dns-setup-disabled": false
    },
    "user-auth": {
        "username": "MyVpnUserName",
        "password": "MyVpnUserPassword"
    }
}
```

**MY QUESTION** is...
**1a)** Is the openvpn3 autoloader compatable with tls-crypt-V2? 
**1b)** If so, what am I missing in my configuration.conf or configuration.autoload files that is required by a tls-crypt-v2 certificate?
**1c)** If I am missing nothing, what other aspects of my configuration could be prohibiting tls-crypt-v2 from working with the autoloader?

All help is appreciated!

**Comment by schwabe** [comment]
The error thrown is from here
```
      if (!SSLLib::PEMAPI::pem_decode(data, key_text.c_str(), key_text.length(),
				      tls_crypt_v2_client_key_name))
	throw tls_crypt_v2_client_key_parse_error();
```

That indicates our OpenVPN 3 core has some problem with the key format in your configuration. Is there anything strange with your key? Or can you generate a key that/config file that you can share with us? It is hard to fix this bug without being to reproduce it.

**Comment by dsommers** [comment]
It's "simpler" than that.  There is a known issue with the `--tls-crypt-v2` option parser in the OpenVPN 3  Python module shipped in openvpn3-linux.  Using `openvpn3` directly (written in C++) should work.  But need to fix this issue in the Python parser.

**Comment by dsommers** [comment]
This should be resolved in commit e83053a954ecaacd7d4bb747884f8a684dbb54ee.  This will be part of the next release (v17_beta)

**Comment by dsommers** [comment]
v17_beta has been released now.
